packages/react-components/src/trace-explorer/trace-explorer-views-widget.tsx

```diff
@@ -23,24 +23,28 @@ export class ReactAvailableViewsWidget extends React.Component<ReactAvailableVie
     static LINE_HEIGHT = 16;
     static ROW_HEIGHT = (2 * ReactAvailableViewsWidget.LINE_HEIGHT) + ReactAvailableViewsWidget.LIST_MARGIN;
 
-    protected forceUpdateKey = false;
-    protected lastSelectedOutputIndex = -1;
-
-    protected selectedExperiment: Experiment | undefined;
+    private _forceUpdateKey = false;
+    private _lastSelectedOutputIndex = -1;
+    private _selectedExperiment: Experiment | undefined;
+    private _experimentManager: ExperimentManager;
 
     constructor(props: ReactAvailableViewsProps) {
         super(props);
-        signalManager().on(Signals.AVAILABLE_OUTPUTS_CHANGED, (payload: AvailableViewsChangedSignalPayload) => this.doHandleAvailableViewsChanged(payload));
+        this._experimentManager = this.props.tspClientProvider.getExperimentManager();
+        this.props.tspClientProvider.addTspClientChangeListener(() => {
+            this._experimentManager = this.props.tspClientProvider.getExperimentManager();
+        });
+        signalManager().on(Signals.EXPERIMENT_SELECTED, (experiment: Experiment | undefined) => this.onExperimentSelected(experiment));
         this.state = { availableOutputDescriptors: [] };
     }
 
     componentWillUnmount(): void {
-        signalManager().off(Signals.AVAILABLE_OUTPUTS_CHANGED, (payload: AvailableViewsChangedSignalPayload) => this.doHandleAvailableViewsChanged(payload));
+        signalManager().off(Signals.EXPERIMENT_SELECTED, (experiment: Experiment | undefined) => this.onExperimentSelected(experiment));
```

Comment:
The listener does not get removed because it is not the same instance being passed that was added earlier.